src/verification/mod.rs

```diff
@@ -353,17 +355,25 @@ where
                                 .join(", ")
                         )));
                     } else if audiences.len() > 1 {
-                        audiences
+                        // first check that the audiences contains the client ID
+                        let found_audience = audiences
                             .iter()
-                            .filter(|aud| (**aud).deref() != self.client_id.deref())
-                            .find(|aud| !(self.other_aud_verifier_fn)(aud))
-                            .map(|aud| {
-                                Err(ClaimsVerificationError::InvalidAudience(format!(
-                                    "`{}` is not a trusted audience",
-                                    **aud,
-                                )))
-                            })
-                            .unwrap_or(Ok(()))?;
+                            .find(|aud| aud.as_str().eq(self.client_id.as_str()));
```

Comment:
First check that the client id is in the provided audience. Only when it isn't found do we apply the other audience verifier.